spring-integration-smb/src/test/java/org/springframework/integration/smb/filters/LastModifiedSmbFileListFilterTests.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2015-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.smb.filters;
+
+import java.util.concurrent.TimeUnit;
+
+import jcifs.smb.SmbException;
+import jcifs.smb.SmbFile;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.file.remote.session.SessionFactory;
+import org.springframework.integration.smb.SmbTestSupport;
+import org.springframework.integration.smb.session.SmbRemoteFileTemplate;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Adama Sorho
+ * @since 6.2
+ */
+@SpringJUnitConfig
+@DirtiesContext
+public class LastModifiedSmbFileListFilterTests extends SmbTestSupport {
```

Comment:
And this one is more drastic than others: it starts a Docker container for Samba server.
Therefore I'd prefer to have all these new tests against mocks and fully in-memory.